package com.example.oops.threading;
class Table{
   public void printtable(int n)  {
       synchronized(this) {
           for (int i = 0; i <= 5; i++) {
               System.out.println(n * i);
               try {
                   Thread.sleep(500);
               } catch (Exception e) {
                   System.out.println(e);
               }
           }
       }
    }
}
class Thread1 extends Thread{



        Table t;
        Thread1(Table t)
        {
            this.t=t;

        }

    @Override
    public void run() {
        t.printtable(5);
    }
}
class Thread2 extends Thread{



    Table t;
    Thread2(Table t)
    {
        this.t=t;

    }

    @Override
    public void run() {
        t.printtable(100);
    }
}

public class SynchronizedBlock {

    public static void main(String[] args) {

        Table t=new Table();

        Thread1 t1=new Thread1(t);
        Thread2 t2=new Thread2(t);
        t1.start();
        t2.start();

    }
}
